romtest/src/main.cpp

```diff
@@ -0,0 +1,59 @@
+#include <fstream>
+#include <stdexcept>
+#include <string>
+
+#include <fmt/core.h>
+
+#include "nes/core/immu.h"
+#include "nes/core/imos6502.h"
+#include "nes/core/ippu.h"
+#include "nes/core/opcode.h"
+#include "nes/disassembler.h"
+#include "nes/nes.h"
+
+namespace {
+
+void print_nametable(const n_e_s::nes::Nes &nes) {
+    for (uint16_t y = 0; y < 30; ++y) {
+        for (uint16_t x = 0; x < 32; ++x) {
+            // $2000-$23FF  Nametable 0
+            // $2400-$27FF  Nametable 1
+            // $2800-$2BFF  Nametable 2
+            // $2C00-$2FFF  Nametable 3
+            const int nametable = 0;
+            const uint16_t address = 0x2000 + 0x0400 * nametable + y * 32 + x;
+            const uint8_t tile_index = nes.ppu_mmu().read_byte(address);
+            fmt::print("{:02X},", tile_index);
+        }
+        fmt::print("\n");
+    }
+}
+
+} // namespace
+
+int main(int argc, char **argv) {
+    if (argc != 3) {
+        fmt::print(stderr, "Expected two argument; <rom.nes> <cycles>\n");
+        return 1;
+    }
+
+    try {
```

Comment:
You could use the function-try-block construction to reduce indentation here. :P 
```
int main(int argc, char **argv) try {
  ...
} catch (std::exception const &e) {
  ...
  return 1;
}
```